src/cata_algo.h

```diff
@@ -2,10 +2,14 @@
 #ifndef CATA_ALGO_H
 #define CATA_ALGO_H
 
+#include <assert.h>
```

Comment:
```suggestion
#include <cassert>
```

"assert.h" is a C header. "cassert" is the matching C++ header.

Reply:
Done.  If you're watching out for those kind of issues I guess it's time for someone to enable the `modernize-deprecated-headers` check in `.clang-tidy` :).